src/OpenFeature.Contrib.ConfigCat/ConfigCatProvider.cs

```diff
@@ -0,0 +1,99 @@
+        /// <param name="configBuilder">The action used to configure the client.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="sdkKey"/> is <see langword="null"/>.</exception>
+        /// <exception cref="ArgumentException"><paramref name="sdkKey"/> is an empty string or in an invalid format.</exception>
+        public ConfigCatProvider(string sdkKey, Action<ConfigCatClientOptions> configBuilder = null)
+        {
+            Client = ConfigCatClient.Get(sdkKey, configBuilder);
+        }
+
+        /// <inheritdoc/>
+        public override Metadata GetMetadata()
+        {
+            return new Metadata(Name);
+        }
+
+        /// <inheritdoc/>
+        public override async Task<ResolutionDetails<bool>> ResolveBooleanValue(string flagKey, bool defaultValue, EvaluationContext context = null)
+        {
+            return await ProcessFlag(flagKey, context, defaultValue);
+        }
+
+        /// <inheritdoc/>
+        public override Task<ResolutionDetails<string>> ResolveStringValue(string flagKey, string defaultValue, EvaluationContext context = null)
+        {
+            return ProcessFlag(flagKey, context, defaultValue);
+        }
+
+        /// <inheritdoc/>
+        public override Task<ResolutionDetails<int>> ResolveIntegerValue(string flagKey, int defaultValue, EvaluationContext context = null)
+        {
+            return ProcessFlag(flagKey, context, defaultValue);
+        }
+
+        /// <inheritdoc/>
+        public override Task<ResolutionDetails<double>> ResolveDoubleValue(string flagKey, double defaultValue, EvaluationContext context = null)
+        {
+            return ProcessFlag(flagKey, context, defaultValue);
+        }
+
+        /// <inheritdoc/>
+        public override async Task<ResolutionDetails<Value>> ResolveStructureValue(string flagKey, Value defaultValue, EvaluationContext context = null)
+        {
+            var stringDefaultValue = defaultValue?.AsString;
+            var result = await ProcessFlag(flagKey, context, stringDefaultValue);
+            var returnValue = result.Value == null ? defaultValue : new Value(result.Value);
+            return new ResolutionDetails<Value>(flagKey, returnValue, variant: result.Variant, errorMessage: result.ErrorMessage);
+        }
+
+        private async Task<ResolutionDetails<T>> ProcessFlag<T>(string flagKey, EvaluationContext context, T defaultValue)
+        {
+            var user = context?.BuildUser();
+            var result = await Client.GetValueDetailsAsync(flagKey, defaultValue, user);
+            return string.IsNullOrEmpty(result.ErrorMessage)
+                ? new ResolutionDetails<T>(flagKey, result.Value, variant: result.VariationId)
+                : new ResolutionDetails<T>(flagKey, defaultValue, ParseErrorType(result.ErrorMessage), errorMessage: result.ErrorMessage);
+        }
+
+        private static ErrorType ParseErrorType(string errorMessage)

```

Comment:
As the error messages can change much easier, I'd go with checking for the error codes and levels:
https://github.com/configcat/.net-sdk/blob/master/src/ConfigCatClient/Logging/LogMessages.cs#L10
The format looks like this: https://github.com/configcat/.net-sdk/blob/master/src/ConfigCat.Client.Tests/data/evaluationlog/simple_value/int_setting.txt#L1

So e.g. instead of "Config JSON is not present", I'd check for "ERROR [1000]"

Reply:
Yeah, I would love to use the error codes, but unfortunately the `ErrorMessage` does not contain it.
I also checked the exception object, and it does not contain the error code.
This code is only used in the Logger.
![image](https://github.com/open-feature/dotnet-sdk-contrib/assets/292532/9e96c23a-6aeb-4251-8ac3-152df05a3501)

Reply:
OMG, I'm really sorry, I totally thought that we are giving back the error code too. 
I'll talk about this with the guys. Totally makes sense to give back the error code too.

Reply:
I guess a new issue can be created if this is added to the config cat SDK.

Reply:
I left it as is for now since it'll capture the general exception anyway.